packages/react-core/src/modules/core/hooks/useStableOptions.ts

```diff
@@ -1,13 +1,13 @@
+import fastDeepEqual from "fast-deep-equal";
```

Comment:
I considered rolling something manually on our own, but decided against it because there's always a risk of a leaky implementation and no guarantee it'd be faster or more performant than existing solutions. 

I did some google search and it seems like this one is what people seem to recommend as the fastest option. I checked the bundle size and it clocks in at ~660b compressed. 

Unclear how this compares to `zustand/shallow` performance wise though, but I suspect it's comparable assuming objects that are not huge or deeply nested.

![CleanShot 2024-11-08 at 15 51 04@2x](https://github.com/user-attachments/assets/6660ff12-d0d3-4655-b0c4-9f2608319430)